registry/remote/credentials/store.go

```diff
@@ -167,6 +169,14 @@ func (ds *DynamicStore) IsAuthConfigured() bool {
 	return ds.config.IsAuthConfigured()
 }
 
+// ConfigPath returns the path to the config file.
+func (ds *DynamicStore) ConfigPath() (string, error) {
+	if ds.config == nil {
+		return "", fmt.Errorf("config is not loaded")
+	}
```

Comment:
Is this part necessary? Is there a valid case where `ds.config` is not loaded? If `ds` is nil, every method will panic.

Reply:
Should check both `ds` and `ds.config` since user can
```golang
var ds *DynamicStore
path, err := ds.ConfigPath()
```
or
```
```golang
var ds DynamicStore
path, err := ds.ConfigPath()
```
```

Reply:
If one does this, the program will panic.
Methods like  `store.Get` and `store.IsAuthConfigured` will also panic.

Reply:
Do we need to tolerate such programming fault?

Reply:
I am not aware that such error are not handled on oras-go side, since we have existing cases for `store.Get` and `store.IsAuthConfigured`, I am happy to remove it.